spine-csharp/src/ExposedList.cs

```diff
@@ -0,0 +1,583 @@
+﻿//
+// System.Collections.Generic.List
+//
+// Authors:
+//    Ben Maurer (bmaurer@ximian.com)
+//    Martin Baulig (martin@ximian.com)
+//    Carlos Alberto Cortez (calberto.cortez@gmail.com)
+//    David Waite (mass@akuma.org)
+//
+// Copyright (C) 2004-2005 Novell, Inc (http://www.novell.com)
+// Copyright (C) 2005 David Waite
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Spine {
+	[Serializable]
+	[DebuggerDisplay("Count={Count}")]
+	public class ExposedList<T> : IEnumerable<T> {
```

Comment:
NGUI uses "BetterList" heh

On Fri, Feb 27, 2015 at 8:02 PM, Serhii Yolkin <notifications@github.com>
wrote:

> In spine-csharp/src/ExposedList.cs
> <https://github.com/EsotericSoftware/spine-runtimes/pull/375#discussion_r25554048>
> :
>
> > +// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
> > +// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
> > +// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
> > +// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
> > +// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
> > +//
> > +
> > +using System;
> > +using System.Collections;
> > +using System.Collections.Generic;
> > +using System.Diagnostics;
> > +
> > +namespace Spine {
> > +	[Serializable]
> > +	[DebuggerDisplay("Count={Count}")]
> > +	public class ExposedList<T> : IEnumerable<T> {
>
> I like Array as a name in general, but I actually use it quite often in
> user code, and it's also confusing because it bears no resemblance with
> regular List, so Array is a personal 'no'. Ideas: ExList, RawList. It's
> actually called NanoList in my other projects, but that's a stupid name :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EsotericSoftware/spine-runtimes/pull/375/files#r25554048>
> .
>